src/server/gc_worker/compaction_filter.rs

```diff
@@ -60,6 +62,7 @@ pub struct GcContext {
     pub(crate) region_info_provider: Arc<dyn RegionInfoProvider + 'static>,
     #[cfg(any(test, feature = "failpoints"))]
     callbacks_on_drop: Vec<Arc<dyn Fn(&WriteCompactionFilter) + Send + Sync>>,
+    pub(crate) tablet_factory: Option<Arc<dyn TabletFactory<RocksEngine> + Send + Sync>>,
```

Comment:
I presume there's no need to keep the `db` field anymore? The single-rocksdb case can also be handled with `KvEngineFactory`.

Reply:
The `db` field represents the global one. And it's needed to support legacy unit cases.

Reply:
You can use `TestTabletFactory`.

Reply:
fixed.